[Nullability Annotations to Java Classes] Add Missing Nullability Annotations to All `Comment` SqlUtils & Model Classes (`breaking`)

ParaskP7 opened the pull request:
Parent #2799
Closes #2831

This PR adds any missing nullability annotation to all `Comment` related sql-util & model classes.

FYI.1: This change is `breaking`, meaning that any client that depended on the `Comment` model should update to the new APIs. As such, this change is inherently `risk`, meaning that there are compile-time changes associated with this change and thus needs testing to verify correctness, both on the library's and client's side.

FYI.2: Comparing to a [previous](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2851) such PR, in this PR there are no `non-default constructors` and `default constructor` changes. This is because of the fact that there are a couple of pseudo `Comment` model construction usages within the JP/WPAndroid repo ([here](https://github.com/wordpress-mobile/WordPress-Android/blob/1f67fab3cec29e0a96423985f5b9118084c246db/WordPress/src/main/java/org/wordpress/android/push/NotificationsProcessingService.java#L628-L630) and [here](https://github.com/wordpress-mobile/WordPress-Android/blob/1f67fab3cec29e0a96423985f5b9118084c246db/WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java#L1149-L1151)), which makes it really hard to re-define and scope the `Comment` model. As such, and in order to avoid causing a larger refactor, adding `non-default constructors` constructors and deprecating the `default constructor` on `Comment` model was ignored.

PS.1: Any other changes on any other class are just some propagating missing nullability annotation changes, which stem from the main set of classes being updated, this PR's focus, plus, some other extra minor analysis, refactor and test related changes.

PS.2: As part of this 3821422ca9f71e4def3455ad021c729feb412680 commit, the [CommentsMapper.kt](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/21b9cb51949bff38713430b86c31e4620623dcf2/fluxc/src/main/java/org/wordpress/android/fluxc/model/comments/CommentsMapper.kt#L4) and its [commentEntityToLegacyModel(...)](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/21b9cb51949bff38713430b86c31e4620623dcf2/fluxc/src/main/java/org/wordpress/android/fluxc/model/comments/CommentsMapper.kt#L54-L76) function got updated to respect the updated nullability contract of the [CommentModel.java](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/21b9cb51949bff38713430b86c31e4620623dcf2/fluxc/src/main/java/org/wordpress/android/fluxc/model/CommentModel.java#L4) model class, which is no longer allowing for a nullable `mStatus`, `mDatePublished`, `mContent` and `mUrl` fields. From my understanding, the author of  [CommentsMapper.kt](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/21b9cb51949bff38713430b86c31e4620623dcf2/fluxc/src/main/java/org/wordpress/android/fluxc/model/comments/CommentsMapper.kt#L4) created those mappings and the [CommentEntity](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/21b9cb51949bff38713430b86c31e4620623dcf2/fluxc/src/main/java/org/wordpress/android/fluxc/persistence/comments/CommentsDao.kt#L276C16-L306) database entity by defaulting all non primitive fields of the `Comment` model to being nullable. But, from my testing, and [this](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2868) related PR (its response) at least 4 of them can't and shouldn't be nullable, as in some cases, if those were to be defined as nullable, that would have caused a NPE. Be extra mindful of that when reviewing this PR.

-----

@wzieba let me know how I did on this PR, that is,  in terms of updating the store/sqlutils classes and adding nullability annotations on them and their main [CommentModel.java](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/21b9cb51949bff38713430b86c31e4620623dcf2/fluxc/src/main/java/org/wordpress/android/fluxc/model/CommentModel.java#L4) model class, in order to utilize the nullability annotations added on its fields and thus make its use null proof. If you also feel that this is a good way to proceed with such PRs, I'll then use this pattern across the board and follow a similar approach on every such PR.

-----

Nullability Annotations List (`Model`):

1. [Add missing n-a to comment model](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/3821422ca9f71e4def3455ad021c729feb412680)

Nullability Annotations List (`SqlUtils`):

1. [Replace collections with cmt list sort on comments sql utils](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/8b52511b295bfc93078a87124db072a692ed4d16)
2. [Replace anonymous classes with lambda on comment sql utils](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/93ece6b5bb17d91789df593b089b8525a30bec9a)
3. [Add missing n-a to insert or update cmt on comment sql utils](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/dce9c685c0f0d60cdeef691251cd55ed13a31fb6)
4. [Add missing n-a to remove comment on comment sql utils](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/3585f82c33a251fe914d478a53024388dd75aea0)
5. [Add missing n-a to remove comments on comment sql utils](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/64a642b804893f22080f2e63afc3bd1d199a6ea7)
6. [Add missing n-a to remove comment gaps on comment sql utils](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/f2b5c5c8d6dc79b4276dc85f2884172538dca88b)
7. [Add missing n-a to insert or update cmt likes on cmt sql utils](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/d24959cdc8eb5e81173b8bae10426433a263be10)
8. [Add missing n-a to get cmt likes by cmt id on cmt sql utils](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/a164dd020a556a647fe3e66e2e5bd64a4d64080f)

Nullability Annotations List (`Store`):

1. [Add missing n-a to comment error on comment store](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/6e7364979bb01c240bd94436235300c4aa4d0b70)
2. [Add missing n-a to on comment changed on comment store](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/97617df3fcb45ba633e7a46f9251dd6dde4a6bff)
3. [Add missing n-a to on comment likes changed on comment store](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/4f45ac862f1a7dc4848a31a4c74c8bf7a0d6e28e)
4. [Add missing n-a to get comments for site on comment store](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/afe04d3cb7ef9d0ad5f45df5036eb7884620c33a)
5. [Add missing n-a to get comment by local id on comment store](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/6f6f78700a3f7b34d38fe2b68c31937cd27c89aa)
6. [Add missing n-a to all payload methods on comment store](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/ad4e4fdb394a42439b3a3a3dc4b785010e650e7d)

Nullability Annotations List (`Utils`):

1. [Add missing n-a to comment error to fetch comments payload](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/0465dffc8bdc156c6d40f43b68035df42824c088)
2. [Add missing n-a to comment error to fetched cmt likes payload](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/3d476284fb5641f93e203dd6acdb73a8e1ddd335)
3. [Add missing n-a to comment error to push comment payload](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/bd52c7237b2166df5ae5def99ac01dee2bef9c4b)

Nullability Checks List:

1. [Guard usages of statuses on comment sql utils](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/e3dfdd553a8bd43c8afdba3c046e58b4ee1a195b)

Warnings Resolution List:

1. [Create missing switch branch on comment error utils](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/dfc9856a23ea7ea75493ea08a106a6d94309c635)
2. [Replace comments published timestamp with long compare](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/fe724caae5b2bd5e9d39d9b097a43b22ec495da2)

Warnings Suppression List:

1. [Suppress resource warning for well sql give me writable db](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/33328c951597dc88609c5bc69d8b14ccc6b583c3)

Refactor List:

1. [Remove unused transient level field from comment model](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/54ffcc1c7e0f3b8c5d58c98fdd91ba7e1495c429)
2. [Replace anonymous classes with lambda on comment sql utils](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/93ece6b5bb17d91789df593b089b8525a30bec9a)

-----

## Associated Clients

It is highly recommending that this change is being tested alongside the below associated clients and their accompanying PRs:

- [JP/WPAndroid#19389](https://github.com/wordpress-mobile/WordPress-Android/pull/19389)

-----

## To Test (`REST`):

- Smoke test the `FluxC Example` app via the `COMMENTS` screen. Verify everything is working as expected.
- In addition to the above, using [local-builds.gradle](https://github.com/wordpress-mobile/WordPress-Android/blob/44a6d0611175f132ff6920b4d6a3e16d565259b6/local-builds.gradle-example#L18) on [JP/WPAndroid](https://github.com/wordpress-mobile/WordPress-Android), smoke test any `CommentStore` and `CommentsStore` related functionality on both, the WordPress and Jetpack apps, and see if everything is working as expected. For a couple of examples, you can expand and follow the inner and more explicit test steps within:

## To Test (`XMLRPC`):

- Create a new self-hosted WordPress site for XMLRPC testing purposes ([jurassic.ninja](https://jurassic.ninja/)).
- Smoke test the `FluxC Example` app via the `COMMENTS` screen. Verify everything is working as expected.
- In addition to the above, using [local-builds.gradle](https://github.com/wordpress-mobile/WordPress-Android/blob/44a6d0611175f132ff6920b4d6a3e16d565259b6/local-builds.gradle-example#L18) on [JP/WPAndroid](https://github.com/wordpress-mobile/WordPress-Android), smoke test any `CommentStore` and `CommentsStore` related functionality on both, the WordPress and Jetpack apps, and see if everything is working as expected. For a couple of examples, you can expand and follow the inner and more explicit test steps within:

<details>
    <summary>1. Comment Details Screen from Comments Screen [CommentDetailFragment.java]</summary>

ℹ️ This test applies to both, the `Jetpack` and `WordPress` app.

- Go to `Comments` screen and tap on any comment.
- Verify that the `Comment Details` screen is shown and functioning as expected.
- For example, try replying, approving, marking as spam, liking, trashing, editing or copying a comment's link address. Additionally, make sure to verify that you can still swipe left-right to navigate between comments.

</details>

<details>
    <summary>2. Comment Details Screen from Notification Tab [CommentDetailFragment.java]</summary>

ℹ️ This test applies to the `Jetpack` app only.

- Go to `Notifications` tab, then its `COMMENTS` sub-tab, and tap on any comment.
- Verify that the `Comment Details` screen is shown and functioning as expected.
- For example, try replying to a comment. Additionally, make sure to verify that you can still swipe left-right to navigate between comments.

</details>

<details>
    <summary>3. Edit Comment Screen from Comments Screen [UnifiedCommentsEditActivity.java]</summary>

ℹ️ This test applies to both, the `Jetpack` and `WordPress` app.

- Go to `Comments` screen and tap on any comment.
- From the `More` on the right side, at the bottom of the comment, select `Edit`.
- Verify that the `Edit Comment` screen is shown and functioning as expected.
- For example, try editing/adding the comment's `Name`, `Web address`, `Email address` and its `Comment` itself.

</details>

<details>
    <summary>4. Edit Comment Screen from Notification Tab [UnifiedCommentsEditActivity.java]</summary>

ℹ️ This test applies to the `Jetpack` app only.

- Go to `Notifications` tab, then its `COMMENTS` sub-tab, and tap on any comment.
- From the `More` on the right side, at the bottom of the comment, select `Edit`.
- Verify that the `Edit Comment` screen is shown and functioning as expected.
- For example, try editing/adding the comment's `Name`, `Web address`, `Email address` and its `Comment` itself.

</details>

<details>
    <summary>5. Reader Comments Screen [ReaderCommentListActivity.java]</summary>

ℹ️ This test applies to the `Jetpack` app only.

- Go to `Reader` tab, then its `DISCOVER` sub-tab, and tap on any post.
- Tap on the `Comments` button at the bottom of the post.
- Verify that the `Reader Comments` screen is shown and functioning as expected.
- For example, try (un)following, enabling/disabling in-app notifications, replying to post, and then sharing, replying or liking a specific comment.

</details>

-----

## Merge instructions:

1. [x] Wait for the accompanying [JP/WPAndroid#19389](https://github.com/wordpress-mobile/WordPress-Android/pull/19389) to be reviewed and approved.
2. [x] Remove the `[PR] Not Ready for Merge` label.
3. [x] Merge this PR.
4. [x] Follow-up with the merge instructions on the accompanying [JP/WPAndroid#19389](https://github.com/wordpress-mobile/WordPress-Android/pull/19389) client PR.

ParaskP7 committed:
Cleanup: Remove unused transient level field from comment model

This transient 'level' field doesn't seem to be used within FluxC or
any client app that might be using the this 'CommentModel' model.

FYI: There is another such definition of a transient 'level' field,
which is actually still in use, but that is on the JP/WPAndroid level
and via the 'ReaderComment' model.

PS: Actually, JP/WPAndroid is the only client that is using this
'CommentModel' model, or 'CommentStore' for that matter.

ParaskP7 committed:
Analysis: Add missing n-a to comment model

FYI: 'n-a' stands for 'nullability annotations'.

PS: This 'NotNullFieldNotInitialized' warning got suppressed because a
table related 'model' class can never have its fields initialized via a
constructor initialization, or otherwise for that matter.

ParaskP7 committed:
Analysis: Add missing n-a to comment error on comment store

FYI: 'n-a' stands for 'nullability annotations'.

ParaskP7 committed:
Refactor: Create missing switch branch on comment error utils

Warning: "'switch' statement on enum type
'org.wordpress.android.fluxc.network.BaseRequest.GenericErrorType'
misses cases: 'TIMEOUT', 'NO_CONNECTION', 'NETWORK_ERROR', 'NOT_FOUND',
'CENSORED', 'SERVER_ERROR', 'INVALID_SSL_CERTIFICATE',
'HTTP_AUTH_ERROR', 'AUTHORIZATION_REQUIRED', 'NOT_AUTHENTICATED',
'PARSE_ERROR' and 'UNKNOWN'"

ParaskP7 committed:
Analysis: Add missing n-a to on comment changed on comment store

FYI: 'n-a' stands for 'nullability annotations'.

ParaskP7 committed:
Analysis: Add missing n-a to on comment likes changed on comment store

FYI: 'n-a' stands for 'nullability annotations'.

ParaskP7 committed:
Analysis: Add missing n-a to get comments for site on comment store

FYI: 'n-a' stands for 'nullability annotations'.

ParaskP7 committed:
Analysis: Add missing n-a to get comment by local id on comment store

FYI: 'n-a' stands for 'nullability annotations'.

ParaskP7 committed:
Analysis: Add missing n-a to all payload methods on comment store

FYI: 'n-a' stands for 'nullability annotations'.

ParaskP7 committed:
Analysis: Add missing n-a to comment error to fetch comments payload

FYI: 'n-a' stands for 'nullability annotations'.

ParaskP7 committed:
Analysis: Add missing n-a to comment error to fetched cmt likes payload

FYI: 'n-a' stands for 'nullability annotations'.

ParaskP7 committed:
Analysis: Add missing n-a to comment error to push comment payload

FYI: 'n-a' stands for 'nullability annotations'.

ParaskP7 committed:
Analysis: Replace collections with cmt list sort on comments sql utils

Warning: "Collections.sort could be replaced with List.sort"

ParaskP7 committed:
Refactor: Replace anonymous classes with lambda on comment sql utils

ParaskP7 committed:
Analysis: Replace comments published timestamp with long compare

Warning: "Expression can be replaced with 'Long.compare'"

ParaskP7 committed:
Analysis: Suppress resource warning for well sql give me writable db

Warning: "'SQLiteDatabase' used without 'try'-with-resources statement"

ParaskP7 committed:
Analysis: Guard usages of statuses on comment sql utils

ParaskP7 committed:
Analysis: Add missing n-a to insert or update cmt on comment sql utils

FYI: 'n-a' stands for 'nullability annotations'.

ParaskP7 committed:
Analysis: Add missing n-a to remove comment on comment sql utils

FYI: 'n-a' stands for 'nullability annotations'.

ParaskP7 committed:
Analysis: Add missing n-a to remove comments on comment sql utils

FYI: 'n-a' stands for 'nullability annotations'.

ParaskP7 committed:
Analysis: Add missing n-a to remove comment gaps on comment sql utils

FYI: 'n-a' stands for 'nullability annotations'.

ParaskP7 committed:
Analysis: Add missing n-a to insert or update cmt likes on cmt sql utils

FYI: 'n-a' stands for 'nullability annotations'.

ParaskP7 committed:
Analysis: Add missing n-a to get cmt likes by cmt id on cmt sql utils

FYI: 'n-a' stands for 'nullability annotations'.

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-FluxC-Android into analysis/add-missing-nullability-annotations-to-all-comment-sqlutils-model-classes

wzieba commented on fluxc/src/main/java/org/wordpress/android/fluxc/utils/CommentErrorUtils.java:
```diff
@@ -51,24 +57,41 @@ public static FetchedCommentLikesResponsePayload commentErrorToFetchedCommentLik
                     errorType = CommentErrorType.INVALID_RESPONSE;
                     break;
+                case TIMEOUT:
+                case NO_CONNECTION:
```
Minor, but it looks like a good candidate for simple `if`

```suggestion
        if (error.isGeneric() && error.type == INVALID_RESPONSE) {
                    errorType = CommentErrorType.INVALID_RESPONSE;
```

ParaskP7 replied:
Thanks for noticing that @wzieba , done here: 1b43bf07c2744abac6a23b324e613fe8df607b16

PS: I didn't apply your commit suggestion as I noticed that the `ommentErrorType.` were missing on the if statement. 😊

wzieba commented on fluxc/src/main/java/org/wordpress/android/fluxc/model/CommentModel.java:
```diff
@@ -23,19 +27,19 @@ public class CommentModel extends Payload<BaseNetworkError> implements Identifia
-    @Column private String mAuthorUrl;
-    @Column private String mAuthorName;
-    @Column private String mAuthorEmail;
+    @Nullable @Column private String mAuthorUrl;
```
(no action) From what I see, annotations didn't change anything in how `CommentModelTable#createStatement` creates table - no `NOT NULL` constraints, so I understand there's no need for database migration 👍

ParaskP7 replied:
This is right @wzieba and thanks for your comment, adding the missing nullability annotation on an `@Table` model wouldn't change anything database related, thus no need for a database migration.

FYI: At some point in the future we might consider to also match the database nullability column constraints to that of the models, but at this point I think it is better to not deal with anything database related and just use these nullability annotation as extra information on how to handle these models correctly, that is, checking or avoid checking for null where appropriate.

wzieba replied:
> at this point I think it is better to not deal with anything database related and just use these nullability annotation as extra information

Totally agree 👍

wzieba commented on fluxc/src/main/java/org/wordpress/android/fluxc/persistence/CommentSqlUtils.java:
```diff
@@ -100,12 +101,18 @@ public static int removeCommentGaps(SiteModel site, List<CommentModel> comments,
+            sourceStatuses = Collections.emptyList();
+        }
         ArrayList<CommentStatus> targetStatuses = new ArrayList<>();
-        if (Arrays.asList(statuses).contains(CommentStatus.ALL)) {
```
(no action) interesting annotation on `Arrays.asList`: `androidx.annotation.RecentlyNonNull`. In theory, all of the annotations you add to FluxC could be `RecentlyNonNull` and `RecentlyNullable` 😄

ParaskP7 replied:
Yea, good question (😅), that's true and I have seen those `@Recent` related annotation for non-null and nullable correspondingly. As such, I did consider that, but then decided that it might not be worth it for us to use `@RecentNonNull` vs `@NonNull`, just to avoid making things more complicated in terms of using those annotation and asking others to use one versus the other. Also, this might make it that bit harder to use static analysis on those (at least [our own](https://github.com/wordpress-mobile/WordPress-Lint-Android/blob/trunk/WordPressLint/src/main/java/org/wordpress/android/lint/MissingNullAnnotationDetector.kt), to be used soon). Unless we have a valid reason of why to use `@Recent` over the normal nullability annotation, I think it is better to avoid it for now.

FYI: My hope with this work is that it will make it easier to convert any such Java class to Kotlin later on, and thus, when that happens, none of those annotations will be relevant anymore... 🤞 

PS: I also remember reading [this](https://android-developers.googleblog.com/2020/03/handling-nullability-in-android-11-and.html) blog post on Android Developer Blog at some point in the past. The why they used such `@Recent` annotation is similar to how one would use the `@Deprecated` annotation, to go back at some point and change the `@Recent` annotation with `@NonNull`/`Nullable` accordingly, which seems to be a bit to much for our case. As such, I don't think it is worth for us to do the same. 🤷 

`Nullability in Android 11`
```
Last month, we released the [Android 11 Developer Preview](https://developer.android.com/preview),
which allows you to test out the new Android 11 SDK. We upgraded a number of annotations in the
SDK from @RecentlyNullable and @RecentlyNonNull to @Nullable and @NonNull (warnings to errors)
and continued to annotate the SDK with more @RecentlyNullable and @RecentlyNonNull annotations
on methods that didn’t have nullability information before.
```

wzieba replied:
TIL, thanks!

wzieba commented on fluxc/src/main/java/org/wordpress/android/fluxc/persistence/CommentSqlUtils.java:
```diff
@@ -307,6 +307,7 @@ public static int insertOrUpdateCommentLikes(
         }
     }
 
+    @NonNull
```
Can you share the motivation behind `@NonNull` here? AFAIK, it _can_ be `null` - we can't assert existence of models in database.

ParaskP7 replied:
Thanks for raising this question @wzieba , I was on the fence about it too. To my understanding anything `WellSql.select(...)` should return a `@NonNull` or a primitive result, at least when that is list related, which worse case scenario should return an empty list. If that is that not true, can you give me an example? 🤔 

Myself, I searched all usages of `WellSql.select(...)` in the project and noticed that most (if not all) of the Java code is using it as `@NonNull`. Example [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/6eee26f191edc73d3934ad5876d4a9b7b1456a9f/fluxc/src/main/java/org/wordpress/android/fluxc/persistence/PostSqlUtils.java#L53-L75).

Then, I did the same for any Kotlin related code that is using it and notice that sometimes the `.firstOrNull()` is used and other time it is not, treating that as `@NonNull` as well, that is, especially when a list is returned (which is the case here). Example [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/6eee26f191edc73d3934ad5876d4a9b7b1456a9f/fluxc/src/main/java/org/wordpress/android/fluxc/persistence/ListSqlUtils.kt#L49-L74).

Let me know your thoughts? 🙏

wzieba replied:
Oh I see, I haven't noticed it's a list. Yes, the list should be empty in case of not finding matching data (`SelectQuery:158`)

ParaskP7 replied:
Awesome, thanks for verifying that for me, I wasn't sure if I am missing something obvious here... 😅

wzieba reviewed (approved):
Solid work @ParaskP7 ! Pre-approving as I haven't found dangerous blockers. Leaving some comments for discussion though.

ParaskP7 committed:
Refactor: Simplify if/switch with if statement on comment error utils

Related PR Comment: https://github.com/wordpress-mobile/
WordPress-FluxC-Android/pull/2870#discussion_r1363993819

ParaskP7 commented:
👋 @wzieba and thanks for reviewing, testing and leaving all those thoughtful comment, very useful indeed! 🙇 ❤️ 🚀 

FYI: I have replied to all of your comments and pushed a [commit](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2870/commits/1b43bf07c2744abac6a23b324e613fe8df607b16) to resolve one of them. Please take a look and let me know. I'll wait merging this PR until all comments are resolved. 🙏

wzieba commented:
Thanks for answering comments and the commit, LGTM :shipit: ! @ParaskP7
